client/src/AppRoot.tsx

```diff
@@ -36,49 +37,72 @@ const AppRoot = () => {
     history.push(routerPaths.root)
   }
 
-  const addElection = (newElection: MailElection) => {
-    const newElections = (mailElections || []).concat([newElection])
-    setMailElections(newElections)
-  }
+  const updateMailElections = useCallback(async () => {
+    try {
+      const { mailElections } = await getMailElections()
+      setMailElections(mailElections)
+    } catch (error) {
+      history.push(routerPaths.root)
+      window.location.reload(false)
+      console.error('getAndSetMailElections failed', error) // eslint-disable-line no-console
+    }
+  }, [history])
+
+  useEffect(() => {
+    updateMailElections()
+  }, [
+    setMailElections,
+    updateMailElections,
+    user,
```

Comment:
Are these extra dependencies actually needed? I don’t see where they’re used.

Reply:
Good call. This is not the right way to do this. Changing.